File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int maxP = 0;
        
        int left = 0;
        int right = 0;
        
        while(right < prices.size()) {
            if(prices[right] > prices[left]) {
                int curP = prices[right] - prices[left];
                maxP = max(maxP,curP);
            } else {
                left = right;
            }
            right++;
        }
        
        return maxP;
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l, r, maxP;
        l = 0;
        r = 0;
        maxP = 0;
        
        while(r < prices.length) {
            if(prices[l] < prices[r]) {
                int profit = prices[r] - prices[l];
                maxP = Math.max(maxP, profit);
            } else {
                l = r;
            }
            r++;
        }
        
        return maxP;
        
    }
}